src/upnp_mrreg.cc

```diff
@@ -131,7 +132,7 @@ void MRRegistrarService::processSubscriptionRequest(const std::unique_ptr<Subscr
         config->getOption(CFG_SERVER_UDN).c_str(),
```

Comment:
So, `UpnpAcceptSubscriptionExt` this calls `genaInitNotifyExt` which calls  `genaInitNotifyCommon`, which says "We take ownership of propertySet and will free it"

So I think we should only free this if we get an error back.

Reply:
I'm sorry, but from the code perspective the above mentioned `propertySet` is a `DOMString` which is `propertySet = ixmlPrintNode((IXML_Node *)PropSet)`.

But contrary to `UpnpActionRequest_set_ActionResult` the content of `PropSet` can be freed right after the call because it is copied to the `DOMString`.

For actionResult there is a `handle_invoke_action` in `soap_device.cc` which does `ixmlDocument_free(actionResultDoc);` So the scan result in #2266 is misleading.

Reply:
Ah, yep, you are right, I skimmed the code to hastily